### PR TITLE
Backport/1.0/2018 08 12

### PR DESCRIPTION
**Pull request by tgraf**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5177)

### CODE REVIEW

**Comment by joestringer** [comment]
This will need the same treatment as the 1.1 backport for the CT map upgrade: https://github.com/cilium/cilium/pull/5176/commits/694ed36b49d63616d1284dea79c7b4193102faa0

**Comment by tgraf** [comment]
Decided in developer meeting that 1.0 <-> 1.2 upgrade and downgrade path is not supported. Dropping connection tracking commits.

**Comment by tgraf** [comment]
@joestringer I've cherry picked the backports from the 1.1 branch. Please review carefully so we have a 2nd pair of eyes.

**Comment by tgraf** [comment]
test-me-please

File: pkg/maps/ctmap/ctmap.go [hunk 1]
```diff
@@ -444,3 +495,58 @@ func Flush(m *bpf.Map, mapName string) int {
...
+
+// checkAndUpgrade determines whether the ctmap on the filesystem has different
+// map properties than this version of Cilium, and if so, removes the map from
+// the filesystem so that a subsequent BPF prog install will recreate it with
+// the correct map properties.
+//
+// Returns true if the map was upgraded.
+func checkAndUpgrade(m *bpf.Map, e CtEndpoint, keySize uint32) bool {
+	desiredMapInfo := &bpf.MapInfo{
+		MapType:   bpf.MapTypeHash,
```
Comment (joestringer) [inline pkg/maps/ctmap/ctmap.go:507]: On newer kernels, this is always an LRU. (Affects all versions with this patch)

https://github.com/cilium/cilium/blob/f8ccf6477f80dbe7c6b83ce03840adec3444bd5a/bpf/bpf_lxc.c#L53-L56

I primarily focused on testing with the dev VM, so didn't catch this earlier.

Comment (tgraf) [reply pkg/maps/ctmap/ctmap.go:507]: OK, seems easy to fix. Is it that we don't have access to the probe result yet?

Comment (joestringer) [reply pkg/maps/ctmap/ctmap.go:507]: Correct. I don't think we have access to the probe result from userspace at all.

Comment (joestringer) [reply pkg/maps/ctmap/ctmap.go:507]: PR is out for the issue I highlighted: #5197

**Pull request closed without merging**